akka-cluster/src/main/scala/akka/cluster/Gossip.scala

```diff
@@ -154,10 +154,9 @@ private[cluster] final case class Gossip(
 
     // 1. merge sets of tombstones
     val mergedTombstones = tombstones ++ that.tombstones
-    val newTombstonedNodes = mergedTombstones.keySet diff that.tombstones.keySet
```

Comment:
No, that was the thing, I also thought that perhaps I needed to do a `merged diff (this intersect that)` but that didn't fix the issue either. Not sure what it is I don't get here.

This is suboptimal if there are many tombstones, because even if they don't change much we need to loop over all of them in the next step.